### PR TITLE
Add regex oov plugin

### PR DESCRIPTION
**Pull request by eiennohito**
1. Analysis now tracks length of created words with a 64-bit bitfield instead of only flags whether a word was created at a bounary
2. OOV plugins can access all other created words at this point for the boundary
3. Add a Regex-based OOV plugin which creates OOVs based on a regular expression-based rule.

### CODE REVIEW

File: src/main/java/com/worksap/nlp/sudachi/RegexOovProvider.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        posId = grammar.getPartOfSpeechId(stringPos);
+        if (posId == -1) {
+            throw new IllegalArgumentException("POS " + stringPos + " was not present in the dictionary");
+        }
+        cost = checkedShort(settings, "cost");
+        leftId = checkedShort(settings, "leftId");
+        rightId = checkedShort(settings, "rightId");
+        pattern = checkPattern(settings.getString("regex"));
+        // force compilation of the pattern here
+        pattern.matcher("test").reset();
```
Comment (kazuma-t) [inline src/main/java/com/worksap/nlp/sudachi/RegexOovProvider.java]: You are forcing the pattern to compile in check_patter, do you need it here as well?

File: src/test/java/com/worksap/nlp/sudachi/RegexOovProviderTest.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.worksap.nlp.sudachi
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class RegexOovProviderTest {
```
Comment (kazuma-t) [inline src/test/java/com/worksap/nlp/sudachi/RegexOovProviderTest.kt:22]: Just to be sure, shouldn't we also test if the pattern matches non-single-byte characters? (To make sure the offset calculation in InputText is correct)

**Review by kazuma-t — CHANGES_REQUESTED** [review]

File: src/test/java/com/worksap/nlp/sudachi/RegexOovProviderTest.kt (hunk 2 above)
Comment (eiennohito) [reply src/test/java/com/worksap/nlp/sudachi/RegexOovProviderTest.kt:22]: Two tests use full-width characters which are non-single byte in utf-8

File: src/main/java/com/worksap/nlp/sudachi/RegexOovProvider.java (hunk 1 above)
Comment (eiennohito) [reply src/main/java/com/worksap/nlp/sudachi/RegexOovProvider.java]: Nice catch, thanks.

**Review by kazuma-t — APPROVED** [review]

**Pull request merged**